js/ui/diagram/ui.diagram.leftpanel.js

```diff
@@ -27,15 +29,9 @@ class DiagramLeftPanel extends Widget {
 
         this._renderAccordion($accordion);
     }
-    _getDataSources() {
-        return this.option("dataSources") || {};
-    }
-    _getCustomShapes() {
-        return this.option("customShapes") || [];
-    }
     _getAccordionDataSource() {
         var result = [];
-        var categories = ShapeCategories.load(this._getCustomShapes().length > 0);
+        var categories = ShapeCategories.load(this._customShapes.length > 0);
```

Comment:
Can we use the options("shapes") itself instead of caching its value in the instance._shapes? Its value is already cached in the options.